src/main/java/server/Server.java

```diff
@@ -11,25 +13,26 @@
 
 public class Server {
     private Integer port;
+    private Logger logger = new Logger();
 
     public Server(Integer port) {
         this.port = port;
     }
 
-    public void start(Map<String, Map<String, RequestHandler>> routes) {
+    public void start(Map<String, Map<String, Handler>> routes) {
         try {
             ServerSocket socket = new ServerSocket(port);
             while (true) {
                 Socket clientSocket = socket.accept();
                 Client client = new Client(clientSocket);
-                RequestParser parser = new RequestParser(client);
-                ResponseSelector selector = new ResponseSelector(routes);
-                ResponseFormatter formatter = new ResponseFormatter();
+                Parser parser = new Parser(client);
```

Comment:
This is a fairly small thing, but ConnectionHandler has both `client` and a `parser` with the same `client`. Do you think it would make sense to remove the `client` state from the `parser` and instead have the `ConnectionHandler` pass the client to `parser#parse`?

As a side note, if you removed the request specific state from the `parser`, I think you could reuse the `parser`, `selector,` and `formatter` instead of instantiating a new one for every request.

Reply:
Yes, I think that would make a lot of sense! Do you think the temporal dependency between `parseRequestLine`, `parseRequestHeaders`, and `parseRequestBody` could create any vulnerability to concurrency bugs?

Reply:
Good question, that's important to think about. Based on the current structure of `Parser`, what do you think?

Reply:
I want to say it's fine, because the dependent methods are private, and the temporal dependencies are honored by the only public method that calls them (`parse`).

Reply:
Agreed